src/common/translate_item.js

```diff
@@ -93,7 +93,62 @@ Zotero.Translate.ItemSaver.prototype = {
 	 */
 	saveItems: async function (items, attachmentCallback, itemsDoneCallback=()=>0) {
 		items = await this._processItems(items);
-	
+
+		// For saving webpages, attach SingleFile page data to attachments
+		let pageContent;
+
+		for (let x = 0; x < items.length; x++) {
+			if (items[x].itemType !== 'webpage') {
+				continue;
+			}
+
+			for (let y = 0; y < items[x].attachments.length; y++) {
+				if (items[x].attachments[y].mimeType !== 'text/html') {
+					continue;
+				}
+
+				// If we have not done so already, inject SingleFile and retrieve
+				// the page content
+				if (!pageContent) {
+
+					// Close the window so that we don't include it in SingleFile capture
+					Zotero.Messaging.sendMessage("progressWindow.close", null);
```

Comment:
I think this won't do. I was saving https://www.nytimes.com/2020/07/22/world/coronavirus-covid-19.html and upon clicking Save (with Embedded Metadata) the progress window appears very shortly, then disappears for about 10s and then reappears again, and this is on a fairly speedy CPU and with plenty of RAM to go around. Does Single-File not provide other mechanisms to hide certain elements by CSS selectors or similar?